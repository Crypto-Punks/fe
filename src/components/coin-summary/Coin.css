.Coin {
    .watched {
        color: gold;
    }

    .unwatched {
        color: white;
    }

    button {
        position: absolute;
        z-index: 1;
        background-color: transparent;
        border: none;
        outline: none;
        transition: 0s;
        font-size: 1.5em;
        

        img {
            width: 2vw;
            border: none;
        }
    }

    .negative {
        color: rgb(214, 64, 64);
    }
          
    .positive {
        color: rgb(17, 212, 17);
    }

      .Name {
        font-size: 10pt;
        margin-left: .5em;
        margin-top: 2em;
        letter-spacing: .4em;
        font-weight: bold;
      }
  
      .Change {
        font-size: 15pt;
        margin-left: 5em;
        margin-top: 0;
        letter-spacing: .4em;
        text-align: right;
      }
}

@media (max-width: 500px) {

    .Coin {

        .Name {
            font-size: 5pt;
            margin-left: 1em;
        }

        .Change {
            font-size: 8pt;
            margin-left: 5em;
            display: flex;
            flex-direction: row;
            align-items: flex-start;
            justify-content: flex-start;
        }
    }
}