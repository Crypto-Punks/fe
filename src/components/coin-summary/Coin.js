import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './Coin.css';

const Coin = ({ item, watchList, handleClick, portfolioInvestedCoins }) => {
  const { id, logo, name, price, changePercent24Hr } = item;

  if(id === 'USD') return (
    <div className={styles.Coin}>
      {renderCoinHtml(logo, name, price, changePercent24Hr)}
    </div>
  );

  return (
    <div className={styles.Coin}>
      {
        handleClick && (portfolioInvestedCoins ? !portfolioInvestedCoins.find(element => element.name === id) : true) &&
        <button
          className={watchList.find(element => element.name === id) ? styles.watched : styles.unwatched}
          onClick={() => handleClick(watchList, id)}>
          ðŸŸŠ</button>
      }
      <Link to={`/detail/${id}`}>
        {renderCoinHtml(logo, name, price, changePercent24Hr)}
      </Link>
    </div>
  );
};

Coin.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    logo: PropTypes.string,
    name: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    changePercent24Hr: PropTypes.string.isRequired,
    special: PropTypes.string.isRequired
  }),
  watchList: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string })),
  handleClick: PropTypes.func,
  portfolioInvestedCoins: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    amount: PropTypes.number.isRequired,
  }))
};

export default Coin;


function renderCoinHtml(logo, name, price, changePercent24Hr) {
  const styleClass = parseFloat(changePercent24Hr) < 0 ? 'negative' : 'positive';
  return (
    <li>
      <img src={logo} alt={name} />
      <p className={styles.Name} >{name}</p>
      <section>
        <p className={styles.Change}> Market Price ${price}</p>
        <p className={`${styles.Change} ${styles[styleClass]}`}
        >{changePercent24Hr}%</p>
      </section>
    </li>
  );
}
