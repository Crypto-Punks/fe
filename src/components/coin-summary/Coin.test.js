import React from 'react';
import { shallow } from 'enzyme';
import Coin from './Coin';

const data = {
  item: {},
};

describe('Coin component', () => {
  it('renders Coin', () => {
    const wrapper = shallow(<Coin { ...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
