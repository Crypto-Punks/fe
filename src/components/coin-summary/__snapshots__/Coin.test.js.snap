// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coin component renders Coin 1`] = `
<div
  className="Coin"
>
  <Link
    to="/detail/undefined"
  >
    <li>
      <img />
      <p
        className="Name"
      />
      <section>
        <p
          className="Change"
        >
           Market Price $
        </p>
        <p
          className="Change positive"
        >
          %
        </p>
      </section>
    </li>
  </Link>
</div>
`;
